imports/wasi_snapshot_preview1/fs.go

```diff
@@ -435,70 +435,38 @@ func fdFilestatSetTimesFn(_ context.Context, mod api.Module, params []uint64) Er
 		return ErrnoBadf
 	}
 
-	// Unchanging a part of time spec while executing utimes is extremely complex to add support for all platforms,
```

Comment:
so the key change here is moving this part outside of wasi preview1 to a place we can re-use it everywhere including wasi-filesystem